src/Native/Unix/configure.cmake

```diff
@@ -467,11 +467,19 @@ check_cxx_source_compiles(
     HAVE_TCP_VAR_H
 )
 
+check_include_files(
+    sys/cdefs.h
+    HAVE_SYS_CDEFS_H)
+
+if (HAVE_SYS_CDEFS_H)
+    set(EXTRA_INCLUDE_FILES "#include <sys/cdefs.h>")
+endif()
+
 # If sys/cdefs is not included on Android, this check will fail because
 # __BEGIN_DECLS is not defined
 check_cxx_source_compiles(
     "
-    #include <sys/cdefs.h>
+    ${EXTRA_INCLUDE_FILES}
```

Comment:
I don't think those definitions are passed into these test compilations by default; I'd need to specifically add it by setting `CMAKE_REQUIRED_DEFINITIONS`, I think. I can experiment with that, it would probably be cleaner.

https://cmake.org/cmake/help/v3.0/module/CheckCXXSourceCompiles.html